parsec/backend/postgresql/user_queries/find.py

```diff
@@ -2,140 +2,150 @@
 
 from pendulum import now as pendulum_now
 from functools import lru_cache
-from typing import Tuple, List
+from typing import Tuple, List, Optional
 
 from parsec.api.protocol import UserID, OrganizationID, HumanHandle
 from parsec.backend.user import HumanFindResultItem
 from parsec.backend.postgresql.utils import Q, q_organization_internal_id, query
 
 
 @lru_cache()
-def _q_factory(query, omit_revoked):
+def _q_factory(query, omit_revoked, offset, limit):
     conditions = []
     if query:
         conditions.append("AND user_id ~* $query")
     if omit_revoked:
         conditions.append("AND (revoked_on IS NULL OR revoked_on > $now)")
     return Q(
         f"""
-SELECT user_id
+SELECT
+    user_id,
+    count(*) OVER() AS full_count
 FROM user_
 WHERE
     organization = { q_organization_internal_id("$organization_id") }
     { " ".join(conditions) }
 ORDER BY user_id
+LIMIT { limit }
```

Comment:
`limit` and `offset` are data provided by the user, even if they are sanitized during deserialization, we shouldn't trust them !

So here we expose ourselves to SQL injection, we must instead pass those variable through parameters.

On top of that, `_q_factory` has a `lru_cache`, so we shouldn't provide parameters that could vary a lot (offset and limit could have any value)